File: 782-jewels-and-stones/jewels-and-stones.js
/**
 * @param {string} jewels
 * @param {string} stones
 * @return {number}
 */
var numJewelsInStones = function (jewels, stones) {
    let map = new Map()
    let sum = 0
    for (let i = 0; i < stones.length; i++) {
        map.set(stones[i], (map.get(stones[i]) || 0) + 1)
    }
    for (let i = 0; i < jewels.length; i++) {
        sum += map.get(jewels[i]) || 0
    }
    return sum
};

File: 782-jewels-and-stones/jewels-and-stones.py
class Solution:
    def numJewelsInStones(self, jewels: str, stones: str) -> int:
        hashMap = {}
        ans = 0
        for i in range(len(stones)):
            hashMap[stones[i]] = hashMap.get(stones[i], 0) + 1
        for i in range(len(jewels)):
            ans += hashMap.get(jewels[i] , 0)
        print(hashMap)
        return ans